ClinGen/clincoded
# Connects to #246 date created

## Testing GDM times

1. Create a new GDM, and follow that to the Record Curation page.
2. Note the GDM’s creation time in the header, at the end of the **Creator** line (this shows the GDM object’s ```date_created``` value, and let a minute or two pass.
3. Now change the OMIM ID. This modifies the GDM.
4. Refresh the Record Curation page. The creation time doesn’t change.
5. Open the GDM as a JSON object by copying’s the GDM‘s UUID from the URL, then going to ```localhost:6543/gdm/*uuid*```. Look for the ```last_modified``` and ```date_created``` fields. ```date_created``` should show the same time as displayed in the header, though in UTC instead of the local time displayed. The ```last_modified``` time should show a few minutes later depending on how much time passed since you created the GDM.
6. Go back to the Record Curation page and add a new PMID.
7. Note that the current time (not the creation time) gets displayed on the **Last edited** line in the header. The **Creator** time still shows the time you made the GDM.
7. Once again, examine the GDM. The ```date_created``` time should be the same as before, but the ```last_modified``` time should be updated.

## Testing group times.

1. Add a group. Minimal information is fine.
2. At the Group Submit Results page, choose **No** from the menu and click the **Return to Record Curation Page**. Back on the Record Curation page, note the date/time under the group you made.
3. Click the **View** link for the group you made.
4. Go to the URL bar and add ```?/format=json``` to it to see the JSON for the group object.
5. Find and note the ```date_created``` and ```last_modified``` times. They’ll be close to each other, though not exact because the resolution of the time is very high.
6. Go back to the Record Curation page and let a minute or two pass.
7. Click **Add new Family to this Group**.
8. Add a basic family and click **Save**.
9. On the Family Submit Results page, choose **No** from the menu, then click **Return to Record Curation Page**.
10. The date/time under the group you created stays the same.
11. Open its JSON object again and note the ```date_created``` and ```last_modified``` times. The ```date_created``` time stays the same as it was before, but the ```last_modified``` time shows the updated time that you added the family.

## Testing family times

1. Open the family as a JSON object in the same way you did for the group. Again, note its ```last_modified``` and ```date_created``` time.
2. Back at the Record Curation page, click the **Edit** link under the family you made.
3. Change anything in the family, and click **Save**.
4. Again, open the JSON object for the family and note and compare its ```last_modified``` and ```date_created``` time.
